little_raft/src/replica.rs

```diff
@@ -396,8 +398,25 @@ where
                     self.match_index.insert(from_id, last_index);
                 } else {
                     // Update information about the peer's logs.
-                    self.next_index
-                        .insert(from_id, self.next_index[&from_id] - 1);
+                    //
+                    // If the mismatch_index is no less than the existing
+                    // next_index, then we know that this rejection is a stray
+                    // out-of-order or duplicate rejection. The reason we know
+                    // that is because mismatch_index set by the follower comes
+                    // from the leader's next_index-1, so we know that
+                    // mismatch_index can't be greater than or equal to
+                    // next_index.
```

Comment:
How about something even more explicit:

> We know this because mismatch_index is set by the follower to prev_log_index, which was in turn set by the leader to `next_index - 1`. Therefore, if mismatch_index is greater than than equal to next_index, the rejection must be out-of-order or stray.

Reply:
Fixed